src/twisted/scripts/_twistd_unix.py

```diff
@@ -439,6 +442,8 @@ def startApplication(self, application):
             uid = process.uid
         if gid is None:
             gid = process.gid
+        if uid is not None and gid is None:
+            gid = pwd.getpwuid(uid).pw_gid
```

Comment:
Please update [the command line help for `--gid`](https://github.com/twisted/twisted/blob/8f34b9b8b5f2a14750e45e636de22e5bfa1044a6/src/twisted/scripts/_twistd_unix.py#L55) so that it mentions that `--gid` is now optional, and that it defaults to the primary group of the user specified by `--uid`.